test/functional/lib/ipfs-request-gateway-redirect.test.js

```diff
@@ -275,43 +386,63 @@ describe('modifyRequest.onBeforeRequest:', function () {
         // or could produce false-positives such as redirection from localhost:5001/ipfs/path to localhost:8080/ipfs/path
         it('should fix localhost Kubo RPC hostname to IP', async function () {
           const request = url2request('http://localhost:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
-          // expectNoRedirect(modifyRequest, request)
-          expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
-            .to.equal('http://127.0.0.1:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
+
+          if (!isManifestV3) {
+            // this is set as a default rule in MV3
+            expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
+              .to.equal('http://127.0.0.1:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
+          }
         })
         it('should be left untouched if localhost Gateway is used', async function () {
           const request = url2request('http://localhost:8080/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
-          await expectNoRedirect(modifyRequest, request)
+
+          await ensureNoRedirect(modifyRequest, request)
         })
         it('should fix 127.0.0.1 Gateway to localhost', async function () {
           const request = url2request('http://127.0.0.1:8080/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
-          // expectNoRedirect(modifyRequest, request)
-          expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
-            .to.equal('http://localhost:8080/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
+
+          if (!isManifestV3) {
+            // this is set as a default rule in MV3.
+            expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
+              .to.equal('http://localhost:8080/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
+          }
         })
         it('should fix 0.0.0.0 to localhost IP API', async function () {
           // https://github.com/ipfs-shipyard/ipfs-companion/issues/867
           const request = url2request('http://0.0.0.0:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
-          expect((await modifyRequest.onBeforeRequest(request)).redirectUrl)
-            .to.equal('http://127.0.0.1:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/')
+
+          ensureCallRedirected({
+            modifiedRequestCallResp: await modifyRequest.onBeforeRequest(request),
+            MV2Expectation: 'http://127.0.0.1:5001/ipfs/QmPhnvn747LqwPYMJmQVorMaGbMSgA7mRRoyyZYz3DoZRQ/',
+            MV3Expectation: {
+              origin: '^https?\\:\\/\\/0\\.0\\.0\\.0',
+              destination: 'http://127.0.0.1'
```

Comment:
:warning: iiuc  rules like this are too greedy and will break people's developer environments as they may run other things on different port.

Can we make the rule more specific, and add rule that touches only  the RPC port?